pkg/secrets/server.go

```diff
@@ -121,11 +122,17 @@ func (s *Server) RetrieveSecret(ctx context.Context, in *secretsgrpc.SecretsRetr
 
 	var resp secretsgrpc.SecretsRetrieveResponse
 
+	if strings.HasSuffix(in.GetName(), "/") {
```

Comment:
Checks like this seem to happen in multiple places, so this check deserves to be split into one or more helper functions. That way the error message can be consistent, the purpose of the check can be clear, and any potential bugs can be fixed in multiple places at once.

Reply:
Additionally, these helpers could perform other functions. They could sanitize the string, trimming or adding slashes, for example. And they can be extended to throw more errors if we decide to enforce restrictions on what constitutes a valid name.

Reply:
Can it be implementing directly in the same file, or we have a utility class for that ?

Reply:
Implementing directly in the same file is fine. You're welcome to create the utility class if you prefer.